src/wallet/wallet.ts

```diff
@@ -1744,68 +1745,81 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
       }
     }
 
+    // insert the data outputs in the beggining of the array, in case it isn't a NFT it will be added at the end of the array
     if (newOptions.isCreateNFT && dataOutputs.length > 0) {
       outputsObj.push(...dataOutputs);
     }
 
     // a. Token amount
-    const addressToUse = newOptions.address || this.getCurrentAddress({ markAsUsed: true }).address;
-    const address = new Address(addressToUse, { network: this.network });
-    if (!address.isValid()) {
-      throw new SendTxError(`Address ${newOptions.address} is not valid.`);
-    }
+    const address = this.validateAddressAndThrow(newOptions.address);
 
     const p2pkhScript = address.getScript();
     outputsObj.push(new Output(amount, p2pkhScript, { tokenData: 1 }));
 
     if (newOptions.createMint) {
       // b. Mint authority
-      const mintAuthorityAddress =
-        newOptions.mintAuthorityAddress || this.getCurrentAddress({ markAsUsed: true }).address;
-      const mintAuthorityAddressObj = new Address(mintAuthorityAddress, { network: this.network });
-      if (!mintAuthorityAddressObj.isValid()) {
-        throw new SendTxError(`Address ${newOptions.mintAuthorityAddress} is not valid.`);
-      }
-
+      const mintAuthorityAddressObj = this.validateAddressAndThrow(newOptions.mintAuthorityAddress);
       const p2pkhMintAuthorityScript = mintAuthorityAddressObj.getScript();
       outputsObj.push(
         new Output(TOKEN_MINT_MASK, p2pkhMintAuthorityScript, { tokenData: AUTHORITY_TOKEN_DATA })
       );
     }
 
+    // 2. Calculate the fee, we are using a simplified fee calculation since in this method we only have outputs related to the token creation
+    // Data outputs are not included in this fee calculation because they have their own fee (0.01 HTR per data output)
+    let fee = 0n;
+    if (tokenInfoVersion === TokenVersion.FEE) {
+      fee = Fee.calculateTokenCreationTxFee([...outputsObj]);
+    }
+
+    const headers: Header[] = [];
+    if (fee > 0) {
+      headers.push(new FeeHeader([{ tokenIndex: 0, amount: fee }]));
+    }
+
+    // 3. Get utxos for HTR
+    const totalAmount = deposit + fee;
+    const { utxos, changeAmount } = await this.getUtxosForAmount(totalAmount, {
+      token: NATIVE_TOKEN_UID,
+    });
+    if (utxos.length === 0) {
+      throw new UtxoError(
+        `No utxos available to fill the request. Token: HTR - Amount: ${totalAmount}.`
+      );
+    }
```

Comment:
`deposit` and `fee` may be zero, so selecting for `totalAmount === 0` may return an empty array.
Is this right?